rsmt2d_test.go

```diff
@@ -7,54 +10,65 @@ import (
 	"github.com/celestiaorg/rsmt2d"
 )
 
-func TestRSMT2D(t *testing.T) {
-	// Size of each share, in bytes
+func TestEdsRepairRoundtripSimple(t *testing.T) {
 	bufferSize := 64
-	// Init new codec
-	codec := rsmt2d.NewLeoRSFF8Codec()
-
-	ones := bytes.Repeat([]byte{1}, bufferSize)
-	twos := bytes.Repeat([]byte{2}, bufferSize)
-	threes := bytes.Repeat([]byte{3}, bufferSize)
-	fours := bytes.Repeat([]byte{4}, bufferSize)
-
-	// Compute parity shares
-	eds, err := rsmt2d.ComputeExtendedDataSquare(
-		[][]byte{
-			ones, twos,
-			threes, fours,
-		},
-		codec,
-		rsmt2d.NewDefaultTree,
-	)
-	if err != nil {
-		t.Errorf("ComputeExtendedDataSquare failed")
+	tests := []struct {
+		name string
+		// Size of each share, in bytes
+		shareSize int
+		codec     rsmt2d.Codec
+	}{
```

Comment:
can you not just range over `codecs`?

Reply:
I could, yes. But now you can change the shareSize here too. Or you can add a different tree with one line etc. Table driven tests are considered best practice in golang.
https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Reply:
It's just easier to extend that test. All tests should be like this ideally.

Reply:
https://github.com/golang/go/wiki/TableDrivenTests

Reply:
Also, in case of a test-failure you would get a better error message that tells you which sub-test fails. If you iterate over the codecs only, it would give you just the line where the assert failed without telling which codec is to blame (I just ran into this issue in the light of #72).